### PR TITLE
Refactoring to make CosmosReadManyRequestOptions not have a base class

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 1]
```diff
@@ -3,23 +3,298 @@
...
  * Specifies the options associated with read many operation
  * in the Azure Cosmos DB database service.
  */
-public final class CosmosReadManyRequestOptions extends CosmosQueryRequestOptionsBase<CosmosReadManyRequestOptions> {
+public final class CosmosReadManyRequestOptions {
+    private final CosmosReadManyRequestOptionsImpl actualRequestOptions;
+
     /**
-     * Instantiates a new read many request options.
+     * Instantiates a new query request options.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:24]: nit: readMany

Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:24]:
```suggestion
     * Instantiates a new read many request options.
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 2]
```diff
@@ -3,23 +3,298 @@
...
      */
     public CosmosReadManyRequestOptions() {
-        super();
+
+        this.actualRequestOptions = new CosmosReadManyRequestOptionsImpl();
+    }
+
+    /**
+     * Instantiates a new query request options.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:32]:
```suggestion
     * Instantiates a new read many request options.
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 3]
```diff
@@ -3,23 +3,298 @@
...
+     */
+    public String getSessionToken() {
+        return this.actualRequestOptions.getSessionToken();
+    }
+
+    /**
+     * Sets the session token for use with session consistency.
+     *
+     * @param sessionToken the session token.
+     * @return the CosmosQueryRequestOptions.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:78]:
```suggestion
     * @return the CosmosReadManyRequestOptions.
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 4]
```diff
@@ -3,23 +3,298 @@
...
+     * stooped. The optional fields may contain serialized index lookup work that
+     * was done but not yet utilized. This avoids redoing the work again in
+     * subsequent continuations and hence improve the query performance. Setting the
+     * maximum continuation size to 1KB, the Azure Cosmos DB service will only
+     * serialize required fields. Starting from 2KB, the Azure Cosmos DB service
+     * would serialize as much as it could fit till it reaches the maximum specified
+     * size.
+     *
+     * @param limitInKb continuation token size limit.
+     * @return the CosmosQueryRequestOptions.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:103]:
```suggestion
     * @return the CosmosReadManyRequestOptions.
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 5]
```diff
@@ -3,23 +3,298 @@
...
+    public int getResponseContinuationTokenLimitInKb() {
+        return this.actualRequestOptions.getResponseContinuationTokenLimitInKb();
     }
+
+    /**
+     * Sets the {@link CosmosEndToEndOperationLatencyPolicyConfig} to be used for the request. If the config is already set
+     *      * on the client, then this will override the client level config for this request
+     *
+     * @param cosmosEndToEndOperationLatencyPolicyConfig the {@link CosmosEndToEndOperationLatencyPolicyConfig}
+     * @return the CosmosQueryRequestOptions
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:128]:
```suggestion
     * @return the CosmosReadManyRequestOptions
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 6]
```diff
@@ -3,23 +3,298 @@
...
+        this.actualRequestOptions.setCosmosEndToEndOperationLatencyPolicyConfig(cosmosEndToEndOperationLatencyPolicyConfig);
+        return this;
+    }
+
+    /**
+     * List of regions to be excluded for the request/retries. Example "East US" or "East US, West US"
+     * These regions will be excluded from the preferred regions list
+     *
+     * @param excludeRegions the regions to exclude
+     * @return the {@link CosmosQueryRequestOptions}
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:140]:
```suggestion
     * @return the {@link CosmosReadManyRequestOptions}
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 7]
```diff
@@ -3,23 +3,298 @@
...
+    public boolean isQueryMetricsEnabled() {
+        return this.actualRequestOptions.isQueryMetricsEnabled();
+    }
+
+    /**
+     * Sets the option to enable/disable getting metrics relating to query execution on item query requests.
+     * By default query metrics are enabled.
+     *
+     * @param queryMetricsEnabled whether to enable or disable query metrics
+     * @return the CosmosQueryRequestOptions.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:171]:
```suggestion
     * @return the CosmosReadManyRequestOptions.
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 8]
```diff
@@ -3,23 +3,298 @@
...
+     */
+    public String getThroughputControlGroupName() {
+        return this.actualRequestOptions.getThroughputControlGroupName();
+    }
+
+    /**
+     * Set the throughput control group name.
+     *
+     * @param throughputControlGroupName The throughput control group name.
+     * @return A {@link CosmosQueryRequestOptions}.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:190]:
```suggestion
     * @return A {@link CosmosReadManyRequestOptions}.
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 9]
```diff
@@ -3,23 +3,298 @@
...
+     * @return the Dedicated Gateway Request Options
+     */
+    public DedicatedGatewayRequestOptions getDedicatedGatewayRequestOptions() {
+        return this.actualRequestOptions.getDedicatedGatewayRequestOptions();
+    }
+
+    /**
+     * Sets the Dedicated Gateway Request Options
+     * @param dedicatedGatewayRequestOptions Dedicated Gateway Request Options
+     * @return the CosmosQueryRequestOptions
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:208]:
```suggestion
     * @return the CosmosReadManyRequestOptions
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 10]
```diff
@@ -3,23 +3,298 @@
...
+    }
+
+    /**
+     * Sets the thresholdForDiagnosticsOnTracer, if latency on query operation is greater than this
+     * diagnostics will be send to open telemetry exporter as events in tracer span of end to end CRUD api.
+     *
+     * Default is 500 ms
+     *
+     * @param thresholdForDiagnosticsOnTracer the latency threshold for diagnostics on tracer.
+     * @return the CosmosQueryRequestOptions
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:234]:
```suggestion
     * @return the CosmosReadManyRequestOptions
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java (hunk 6 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java (hunk 8 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java [hunk 11]
```diff
@@ -3,23 +3,298 @@
...
+        return this;
+    }
+
+    /**
+     * Gets indexMetricsEnabled, which is used to obtain the index metrics to understand how the query engine used existing
+     * indexes and could use potential new indexes.
+     * The results will be displayed in QueryMetrics. Please note that this options will incurs overhead, so it should be
+     * enabled when debuging slow queries.
+     *
+     * @return indexMetricsEnabled (default: false)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:260]:
```suggestion
     * @return the CosmosReadManyRequestOptions
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java (hunk 9 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java (hunk 11 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java:260]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java (hunk 10 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java (hunk 7 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosReadManyRequestOptions.java]: Fixed

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Pull request merged**